__tests__/app.test.js

```diff
@@ -65,6 +65,60 @@ describe('GET /api/articles', () => {
         .expect(404)
 
     });
+    it('returns 200 status and queried articles but topic when endpoint contains a topic query', () => {
+        return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles[0]).toHaveProperty('topic', 'cats')
+        })
+    });
+    it('returns 200 status and sorts articles by sortBy when endpoint contains a sortBy query ', () => {
+        return request(app)
+        .get("/api/articles?sort_by=author")
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles).toBeSortedBy("author", {
+                descending: true,
+            });
+        });
+    });
+    it('returns 200 status and orders created_by by ascending', () => {
+        return request(app)
+        .get("/api/articles?order=asc")
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles).toBeSortedBy("created_at", {
+                descending: false,
+            });
+        });
+    });
+    it('returns 200 status and sorts articles by author in ascending order', () => {
+        return request(app)
+        .get("/api/articles?sort_by=author&order=asc")
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles).toBeSortedBy("author", {
+                descending: false,
+            });
+        });
+    });
+    it('returns 404 and message if topic category doesn\'t exist', () => {
+        return request(app)
+        .get("/api/articles?topic=stuff")
+        .expect(404)
+        .then(({body}) => {
+            expect(body.message).toBe('Article doesn\'t exist');
+        })
+    });
+    it('returns status of 400 and error message when passed an invalid query ', () => {
+        return request(app)
+        .get("/api/articles?sort_by=stuff")
+        .expect(400)
+        .then(({body}) => {
+            expect(body.message).toBe('Invalid sort query');
+        })
+    });
 
```

Comment:
Another 400 for invalid order i.e. ?order=bananas
Another200 for topic exists but no comments i.e. ?topic=paper

Reply:
I'm not seeing the second test I suggested?